src/main/java/org/opencadc/tap/impl/AuthenticatorImpl.java

```diff
@@ -44,6 +45,20 @@
 
     private static final HttpClient client = HttpClient.newHttpClient();
 
+    private static final HashMap<String,TokenInfo> tokenCache = new HashMap<>();
```

Comment:
`HashMap` is not thraed-safe, so you can't use a global version that would be shared across threads. I think you can just replace it with `ConcurrentHashMap`.